src/net/sourceforge/kolmafia/session/ResultProcessor.java

```diff
@@ -2066,7 +2066,8 @@ private static void gainItem(boolean adventureResults, AdventureResult result) {
         break;
 
       case ItemPool.GOBLIN_WATER:
-        if (adventureResults) {
+        if (adventureResults
+            && KoLCharacter.inRaincore()) { // because you can now get the Goblin Water otherwise...
```

Comment:
What if you are wearing Mr. Cheeng's spectacles in Heavy Rains? :)

Reply:
Then it would incorrectly mark the quest completed.

If I were to solve for that case, I would have to think about a better solution for longer than I thought about this one. 

The reports of the problem were coming from Casual, and I thought the hole would be at least smaller.  I would happily implement a better solution.